### PR TITLE
Implemented #2076 plus extras

### PR DESCRIPTION
**Pull request by spacebuilder2020**
Added @exp argument, as well as an expCheck and expSet method.

Also, added hungerCheck, hungerSet, saturationCheck, and saturationSet
to correspond with existing annotations.

### CODE REVIEW

File: src/main/java/com/forgeessentials/scripting/ScriptMethods.java [hunk 1]
```diff
@@ -502,7 +502,233 @@ public String getHelp()
...
+					return false;
+			}
+			else
+				return false;
+        }
+
+        @Override
+        public String getHelp()
+        {
+            return "`expCheck <expLevel> [operator -- LT, LTE, GT, GTE, or EQ]`  \nThis method will check the expLevel of a player and compare it to a given value. If no operator is specified, EQ will be used as default";
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/scripting/ScriptMethods.java]: People are idiots, what do the operators mean?

Or you can just use `<`, `=<`, `>`, `>=`, and `=`. Not sure how that would work out though.

**Comment by spacebuilder2020** [comment]
The operators allow you to perform comparisons between an existing value and a specified value.  For example being able to determine if a players exp is less than (LT) an inputted value.  Or if it is greater than (GT).  And I chose GT, LT, LTE instead of >, <, <= because GT is easier to read than > for a non programmer.  But I suppose it is up to interpretation.

**Comment by luacs1998** [comment]
Perhaps you should include it in the docs.

**Comment by spacebuilder2020** [comment]
Will Do.  Also, I am going to go ahead improve the code with addon request later today to enable case insensitive LT, GT, LTE, etc and  <, =<, <, etc operators in place of their corresponding operator.

Also, think I am going to add some adder methods when necessary since there is no mathematical operations supported in scripting.  expAdd, saturationAdd, hungerAdd, etc. 

In addition I am thinking of adding chekers, setters, and adders for existing annotations when relevant. (player status, gamemode, etc)

Finally, a @random annotation has been asked for in some form or another in places so I might add that as well.

**Comment by exp** [comment]
FWIW @spacebuilder2020 by using the @ form you hilight people with those usernames ;)

**Comment by spacebuilder2020** [comment]
Sorry... Will find different characters for forge essentials annotations.

**Comment by exp** [comment]
I don't think it matters in the code, just in the issues. You can quote yourself and do this: `@spacebuilder2020` without hilighting.

**Comment by spacebuilder2020** [comment]
Yea, I didn't realize that the mentions I used actually mapped to a user.  You see when identifying scripting arguments, the @ symbol is used both in syntax and documentation, the @ prefix is used to mark that tag as an argument. So I will try to use a substitute character in place of @ when referring to arguments from now on.  Sorry for mentioning you.  It was unintentional.

**Comment by spacebuilder2020** [comment]
Or format it as code if that works.  😊

**Pull request merged**